src/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useMemo } from 'react'
+import * as d3 from 'd3'
+
+import { useD3 } from '../hooks'
+import { DiscreteItem } from '../profiler/model'
+import { COLORS } from '../profiler/constants'
+
+const BarChart = ({ dataset, config }: BarChartProps): JSX.Element => {
+  const { height, color, showAxes, showLogScale } = config
+
+  const margin = useMemo(
+    () => ({
+      left: showAxes ? 25 : 2,
+      right: showAxes ? 25 : 2,
+      top: showAxes ? 25 : 2,
+      bottom: showAxes ? 25 : 2,
```

Comment:
Effectively you need only one margin constant

`const margin = showAxes ? 25 : 2`